bgpd/bgp_clist.c

```diff
@@ -1123,12 +1125,27 @@ static bool lcommunity_list_valid(const char *community)
 		frrstr_split(communities[j], ":", &splits, &num);
 
 		for (int i = 0; i < num; i++) {
-			if (strtoul(splits[i], NULL, 10) > UINT_MAX)
-				return false;
-
 			if (strlen(splits[i]) == 0)
+				/* There is no digit to check */
 				return false;
 
+			if (style == LARGE_COMMUNITY_LIST_STANDARD) {
+				if (strtoul(splits[i], &endptr, 10) > UINT_MAX)
+					/* Larger than 4 octets */
+					return false;
+
+				if (*endptr)
+					/* Not all characters were digits */
+					return false;
+			} else {
+				if (!(regres = bgp_regcomp(communities[j]))) {
```

Comment:
It's enough to call `if (!bgp_regcomp(...))` or do you need somewhy an additional pointer here?

Reply:
`bgp_regcomp` returns a pointer to allocated memory which needs to be released before moving on